src/lib/auth.js

```diff
@@ -38,10 +39,17 @@ class Auth {
 
     const allow_action = function(action) {
```

Comment:
What you are trying to achieve won't work in the way you are expecting for. The reasons:

You have updated the method `allow_action` which is few lines below pushed as a plugin. What that even means? 
```
 this.plugins.push({
      authenticate: function(user, password, cb) {
        return cb( Error[403]('bad username/password, access denied') );
      },

      add_user: function(user, password, cb) {
        return cb( Error[409]('registration is disabled') );
      },

      allow_access: allow_action('access'),
      allow_publish: allow_action('publish'),
    });
```
`verdaccio` includes a built-in auth plugin called `htpasswd`. 
```
auth:
  htpasswd:
    file: ./htpasswd
```
If that part is clear, then we go to the next step. The code you have the update **will never be executed unless there plugin auth in use must not implement** `allow_access`, `allow_publish` or both. That's the case of the default plugin. If you wonder why the tests are failing, keep going.

**Why are the tests failing?**

We trigger 3 `verdaccio` servers (https://github.com/verdaccio/verdaccio/blob/master/test/functional/index.func.js#L53) in the background to run the tests, if you see in detail each of those 3 config files we have defined for test (https://github.com/verdaccio/verdaccio/blob/master/test/functional/store/config-1.yaml#L9) we use the `verdaccio-auth-memory` [plugin](https://github.com/verdaccio/verdaccio-auth-memory), and that plugin implements `allow_access` and `allow_publish` already, so, your code is not reached at all. It cannot be tested with functional testing. 

**I'd suggest use unit test or try to trigger a four server that does not use any auth plugin.**